### PR TITLE
[WNMGDS-723] Add support for CMSDS Typescript components

### PR DESCRIPTION
**Pull request by bernardwang**
### Summary
- Updates deps and scripts to allow for core CMSDS components to be written in Typescript
- Updates `FieldContainer` to typescript

### How to test
- Ensure `FieldContainer` type definition files are generated when DS is built
- Test `FieldContainer` type definition

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -11,10 +13,13 @@ module.exports = function (api) {
         // modules: false
       },
     ],
-    '@babel/preset-react',
   ];
 
-  const plugins = ['@babel/plugin-transform-object-assign', 'inline-react-svg'];
+  const plugins = [
+    '@babel/plugin-proposal-class-properties',
```
Comment (bernardwang) [inline babel.config.js:19]: Updates our default supported plugins to use `@babel/plugin-proposal-class-properties` and `@babel/plugin-proposal-object-rest-spread`

File: packages/design-system-scripts/gulp/build.js [hunk 2]
```diff
@@ -82,37 +80,6 @@ async function copyAll(dir, options) {
...
-        })
-      )
-      .pipe(gulp.dest(path.join(dir, 'dist', 'types')))
-      .on('finish', function () {
-        logTask('📜 ', 'Core Typescript definition files generated');
-      })
-  );
-}
-/* eslint-enable */
-
```
Comment (nichia) [inline packages/design-system-scripts/gulp/build.js:115]: Can you talk more as to why this function is no longer needed?

File: .eslintrc.js [hunk 3]
```diff
@@ -1,8 +1,16 @@
...
     'react/no-danger': 'error',
   },
+  overrides: [
+    {
+      files: ['*.ts', '*.tsx'],
+      parser: '@typescript-eslint/parser',
+      extends: ['plugin:@typescript-eslint/recommended'],
+      rules: {},
+    },
+  ],
```
Comment (nichia) [inline .eslintrc.js:15]: Not sure if it's related to this, but `yarn lint` command fails with this error:
```
[08:42:06] 🔎  Linting files in: ./ and ignoring paths: **/node_modules/**, **/dist/**, **/helpers/**, **/__tests__/**, tmp/**, **/types/**, examples/create-react-app-typescript/**

events.js:292
      throw er; // Unhandled 'error' event
```

**Comment by nichia** [comment]
Great work Bernard in adding typescript support to the CMSDS. 
Tested the forms components on core ds, child ds and the consuming applications and they work as expected.

File: packages/design-system-scripts/gulp/build.js (hunk 2 above)
Comment (bernardwang) [reply packages/design-system-scripts/gulp/build.js:115]: Originally it was used to generate the Typescript definition files automatically, but now we are manually updating them, so its not being used anywhere anyway

File: .eslintrc.js (hunk 3 above)
Comment (bernardwang) [reply .eslintrc.js:15]: i’ll take a look at this soon

**Review by line47 — APPROVED** [review]
This is working for me!

**Comment by bernardwang** [comment]
@nichia the `yarn lint` issue was fixed. There is still an eslint error from the new typescript component but that can be addressed in a followup

**Review by nichia — APPROVED** [review]

**Pull request merged**